content/guides/webpack-and-typescript.md

```diff
@@ -0,0 +1,198 @@
+ output: {
+   filename: '/bundle.js',
+   path: '/'
+ },
+ module: {
+   rules: [
+     {
+       test: /\.tsx?$/,
+       loader: 'ts-loader',
+       exclude: /node_modules/,
+     }
+   ]
+ }
+};
+```
+ 
+Here we specify our entry point to be index.ts in our current directory, 
+an output file called bundle.js 
+and our typescript loader that is in charge of compiling our typescript file to javascript.
+
+## Typescript loaders
+
+Currently there are 2 loaders for typescript available:
+* [awesome-typescript-loader](https://github.com/s-panferov/awesome-typescript-loader)
+* [ts-loader](https://github.com/TypeStrong/ts-loader)
+
+Awesome typescript loader has created a wonderful explanation of the 
+difference between awesome-typescript-loader and ts-loader. 
+
+You can read more about it [here](https://github.com/s-panferov/awesome-typescript-loader#differences-between-ts-loader).
+
+In this guide we will be using ts-loader as currently it is easier enabling additional webpack features such as importing non code assets into your project.
+
+## Enabling source maps
+
+In order to enable source maps we first must configure typescript to output inline source maps to our compiled javascript files.
+This is done by setting the sourceMap property to true.
+ 
+__tsconfig.json__
+```json
+{
+  "sourceMap": true
+}
+```
+
+Once typescript is configured to output source maps we need to tell webpack 
+to extract these source maps and pass them to the browser, this way we will get the source file
+exactly as we see it in our code editor.
+
+__webpack.config.js__
+```js
+module.exports = {
+ entry: './index.ts',
+ output: {
+   filename: '/bundle.js',
+   path: '/'
+ },
+ module: {
+   rules: [
+     {
+       enforce: 'pre',
+       test: /\.js$/,
+       loader: "source-map-loader"
+     },
+   ]
+ },
+ devtool: 'inline-source-map',
+};
+```
+ 
+First we add a new loader called source-map-loader. 
+
+To install it run: 
+
+`npm install --save-dev source-map-loader`.
+
+Once the loader is installed we need to tell webpack we want to run this loader before any other loaders by using the `enforce: 'pre'` configuration flag.
+Finally we need to enable source maps in webpack by specifying the `devtool` property.
+Currently we use the 'inline-source-map' setting, to read more about this settings and see other options check out the [devtool documentation](https://webpack.js.org/configuration/devtool/).
+
+ 
+ 
+ 
+
+## Using 3rd Party Libraries
+
+When installing 3rd party libraries from npm, it is important to remember
+to install the typing definition for that library.
+
+You can install 3rd party library definitions from the @types repository.
+
+For example if we want to install lodash we can run the following command to get the typings for it:
+`npm install --save-dev @types/lodash`
+
+For more information see [this blog post](https://blogs.msdn.microsoft.com/typescript/2016/06/15/the-future-of-declaration-files/)
+
+## Importing non code assets
```

Comment:
I believe that at least the typescript related content in this section should stay in this page, it was a big issue for me learning to use webpack with typescript and there wasn't allot of resources on this specific topic.

Would you agree if I move the part that explains how to load non code assets which means the text until "Here we instruct webpack to include this file in our build and return a url reference in our Typescript code."

And then keep the rest of the content in this page, so people know how to configure typescript to work with custom file loaders?

Also I'm not sure where should the importing non code assets section can go to as I don't see anything existing right now, should it be it's own page?